Marlin/Marlin_main.cpp

```diff
@@ -4701,7 +4697,22 @@ inline void gcode_M109() {
 
     LCD_MESSAGEPGM(MSG_BED_HEATING);
     bool no_wait_for_cooling = code_seen('S');
-    if (no_wait_for_cooling || code_seen('R')) thermalManager.setTargetBed(code_value_temp_abs());
+    if (no_wait_for_cooling || code_seen('R')) {
+      thermalManager.setTargetBed(code_value_temp_abs());
+      #if ENABLED(PRINTJOB_TIMER_AUTOSTART)
+        if(code_value_temp_abs() > BED_MINTEMP) {
```

Comment:
This should produce slightly more optimal code, with one less call to `code_value_temp_abs`:
```cpp
const float temp = code_value_temp_abs();
thermalManager.setTargetBed(temp);
#if ENABLED(PRINTJOB_TIMER_AUTOSTART)
  if (temp > BED_MINTEMP) {
    . . .
```